packages/miniflare/src/plugins/containers/service/index.ts

```diff
@@ -79,4 +95,24 @@ export class ContainerController {
 			);
 		}
 	}
+
+	// TODO should this live in containers-shared?
+	async pullImage(image: string) {
+		try {
+			await dockerLoginManagedRegistry(this.#sharedOptions.dockerPath);
+			await runDockerCmd(this.#sharedOptions.dockerPath, ["pull", image]);
+			// const tag = `${MF_CONTAINER_PREFIX}/${options.name}`;
+			// this needs to be tested cause we might have to extract the actual image tag
+			// await runDockerCmd(this.#sharedOptions.dockerPath, ["tag", image, tag]);
+			this.#logger.info(`Successfully pulled image: ${image}`);
+		} catch (error) {
+			if (error instanceof Error) {
+				throw new Error(error.message);
+			}
```

Comment:
without the following change, the stack trace will be gone.

```suggestion
			if (Error.isError(error)) {
				throw new Error(error.message, { cause: error );
			}
```

Reply:
TIL 🤯 . But also, this is only partially supported starting node `v24`, which was released just a few weeks ago. wrangler needs to support lower node versions `>=18.20.0`